packages/interactivity/src/directives.js

```diff
@@ -267,15 +267,71 @@ export default () => {
 	// data-wp-watch--[name]
 	directive( 'watch', ( { directives: { watch }, evaluate } ) => {
 		watch.forEach( ( entry ) => {
-			useWatch( () => evaluate( entry ) );
+			useWatch( () => {
+				if ( SCRIPT_DEBUG ) {
```

Comment:
@sirreal is trying to advance the proposal https://github.com/WordPress/gutenberg/pull/61486 to unify the usage on constants and prefixing them with `globalThis`:
```suggestion
				if ( globalThis.SCRIPT_DEBUG ) {
```

The benefit would be that the code will continue to work without errors when the `SCRIPT_DEBUG` is not defined or not replaced by the bundler. Currently, projects outside of WordPress need to take care of it, and when they don't this conditional will throw an error.

Reply:
switched here and across other usages